api/v1alpha1/api.go

```diff
@@ -144,3 +150,128 @@ const (
+	// Type specifies the type of the API key. Currently, "SecretRef" is supported.
+	// This defaults to "SecretRef".
+	//
+	// +kubebuilder:validation:Enum=SecretRef
+	// +kubebuilder:default=SecretRef
+	Type ProviderAPIKeyType `json:"type"`
+
+	// SecretRef is the reference to the secret containing the API key.
+	// ai-gateway must be given the permission to read this secret.
+	// The key of the secret should be "apiKey".
+	//
+	// +optional
+	SecretRef *gwapiv1.SecretObjectReference `json:"secretRef"`
+}
+
+// ProviderAPIKeyType specifies the type of ProviderAPIKey.
+type ProviderAPIKeyType string
+
+// AWSSecurityPolicy contains the supported authentication mechanisms to access aws
+type AWSSecurityPolicy struct {
+	// Region specifies the AWS region associated with the policy
+	Region string `json:"region"`
+
+	// CredentialsFile specifies the credentials file to use for the AWS provider.
+	// +optional
+	CredentialsFile *ProviderAWSCredentialsFile `json:"credentialsFile,omitempty"`
+
+	// OIDCCredential specifies the oidc and credentials to use for the AWS provider.
+	// +optional
+	OIDCCredential *ProviderAWSOIDCCredential `json:"oidcCredential,omitempty"`
+}
+
+// ProviderAWSCredentialsFile specifies the credentials file to use for the AWS provider.
+// Envoy reads the credentials from the file pointed by the Path field, and the profile to use is specified by the Profile field.
+type ProviderAWSCredentialsFile struct {
+	// Path is the path to the credentials file.
+	//
+	// +kubebuilder:default=~/.aws/credentials
+	Path string `json:"path,omitempty"`
+
+	// Profile is the profile to use in the credentials file.
+	//
+	// +kubebuilder:default=default
+	Profile string `json:"profile,omitempty"`
+}
+
+// ProviderAWSOIDCCredential specifies credentials to obtain oidc token from a sso server.
+// For AWS, the controller will query STS to obtain AWS AccessKeyId, SecretAccessKey, and SessionToken,
+// and store them in a temporary credentials file
+type ProviderAWSOIDCCredential struct {
+	// OIDC is used to obtain oidc tokens via an SSO server which will be used to exchange for temporary AWS credentials
+	OIDC egv1a1.OIDC `json:"oidc"`
+
+	// CredentialFileName is the name of the configmap that will store the temporary credentials
+	// file for the actual credentials will be `credentials`
+	CredentialFileName string `json:"credentialFileName"`
+
+	// GrantType is the method application gets access token
+	//
+	// +optional
+	GrantType string `json:"grantType,omitempty"`
+
+	// Aud defines the resource the application can access
+	//
+	// +optional
+	Aud string `json:"aud,omitempty"`
+
+	// AwsRoleArn is the identifier for which role the user can access
+	//
+	// +optional
+	AwsRoleArn string `json:"awsRoleArn,omitempty"`
+
+	// AuthBearToken is an optional authorization token that can be passed with OIDC request
+	AuthBearToken string `json:"authBearToken,omitempty"`
+
+	// SSORequestMethod specifies the expected http method for an SSO server
```

Comment:
SSO is related to OIDC but they are different, OIDC is more specific and enables SSO functionality by providing  away to verify a user's identity and share user information. Essentially OIDC is an implementation of SSO.